File: app/src/main/java/com/ponkan/banana/camera/CameraActivity.java
package com.ponkan.banana.camera;

import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;

import com.ponkan.banana.FullScreenActivity;
import com.ponkan.banana.R;

public class CameraActivity extends FullScreenActivity implements OnFragmentInteractionListener {


    private CameraFragment cameraFragment;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_camera);

        if (cameraFragment == null) {
            cameraFragment = CameraFragment.newInstance("heng", "ha");
            getSupportFragmentManager().beginTransaction().add(android.R.id.content, cameraFragment,
                    "cameraFragment").commitAllowingStateLoss();
        }
    }


    @Override
    public void onFragmentInteraction(Uri uri) {

    }

    @Override
    public void onWindowFocusChanged(boolean hasFocus) {
        super.onWindowFocusChanged(hasFocus);
        if (hasFocus) {
            initFullScreen();
        }
    }

    @Override
    public void go2ImagePreview(String imagePath) {
        ImagePreviewFragment imagePreviewFragment = ImagePreviewFragment.newInstance(imagePath, "");
        getSupportFragmentManager().beginTransaction().add(android.R.id.content, imagePreviewFragment,
                "imagePreviewFragment").addToBackStack(null).commitAllowingStateLoss();

    }

    @Override
    public void onBackPressed() {

        // 后退处理，配合addToBackStack使用
        int count = getSupportFragmentManager().getBackStackEntryCount();

        if (count == 0) {
            super.onBackPressed();
        } else {
            getSupportFragmentManager().popBackStack();
        }
    }
}

File: app/src/main/java/com/ponkan/banana/camera/CameraActivity.kt
package com.ponkan.banana.camera

import android.net.Uri
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View

import com.ponkan.banana.R
import android.view.View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY
import android.view.View.SYSTEM_UI_FLAG_FULLSCREEN
import android.view.View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
import android.view.View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
import android.view.View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
import android.view.View.SYSTEM_UI_FLAG_LAYOUT_STABLE



class CameraActivity : AppCompatActivity(), CameraFragment.OnFragmentInteractionListener {

    private var cameraFragment: CameraFragment? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_camera)

        if (cameraFragment == null) {
            cameraFragment = CameraFragment.newInstance("heng", "ha")
            supportFragmentManager.beginTransaction().add(android.R.id.content, cameraFragment, "cameraFragment").commitAllowingStateLoss()
        }
    }

    override fun onFragmentInteraction(uri: Uri) {

    }

    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        val mDecorView = window.decorView
        mDecorView.systemUiVisibility = (View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                or View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                or View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
//                or View.SYSTEM_UI_FLAG_HIDE_NAVIGATION // hide nav bar
                or View.SYSTEM_UI_FLAG_FULLSCREEN // hide status bar
                or View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY)
    }
}
